test/test_transforms_task.py

```diff
@@ -713,3 +714,99 @@ def test_treeherder_defaults(run_transform, graph_config, kind, task_def, expect
     pprint(task_dict, indent=2)
 
     assert task_dict["task"].get("extra", {}).get("treeherder", {}) == expected_th
+
+
+@pytest.mark.parametrize(
+    "no_exception_tasks",
+    (
+        pytest.param(
+            {
+                "label": "task1",
+                "dependencies": ["dependency"] * 2,
+                "soft-dependencies": ["dependency"] * 1,
+                "if-dependencies": ["dependency"] * 4,
+            },
+        ),
+        pytest.param(
+            {
+                "label": "task2",
+                "dependencies": ["dependency"] * 97,
+                "soft-dependencies": ["dependency"],
+                "if-dependencies": ["dependency"],
+            },
+        ),
+    ),
+)
+@pytest.mark.parametrize(
+    "exception_tasks",
+    (
+        pytest.param(
+            {
+                "label": "task3",
+                "dependencies": ["dependency"] * 98,
+                "soft-dependencies": ["dependency"],
+                "if-dependencies": ["dependency"],
+            },
+        ),
+        pytest.param(
+            {
+                "label": "task4",
+                "dependencies": ["dependency"] * 99,
+                "soft-dependencies": ["dependency"],
+                "if-dependencies": ["dependency"],
+            },
+        ),
+    ),
+)
+@pytest.mark.parametrize("noexception", (does_not_raise(),))
+@pytest.mark.parametrize("exception", (pytest.raises(Exception),))
```

Comment:
Nice! The test looks much easier to understand. Today, I learned we can use `@pytest.mark.parametrize` multiple times for a given test. For anyone interested, here's what it does: 

```
[task 2023-05-17T08:46:45.461Z] test/test_transforms_task.py::test_check_task_dependencies[exception0-noexception0-exception_tasks0-no_exception_tasks0] PASSED [ 51%]
[task 2023-05-17T08:46:45.461Z] test/test_transforms_task.py::test_check_task_dependencies[exception0-noexception0-exception_tasks0-no_exception_tasks1] PASSED [ 51%]
[task 2023-05-17T08:46:45.462Z] test/test_transforms_task.py::test_check_task_dependencies[exception0-noexception0-exception_tasks1-no_exception_tasks0] PASSED [ 52%]
[task 2023-05-17T08:46:45.462Z] test/test_transforms_task.py::test_check_task_dependencies[exception0-noexception0-exception_tasks1-no_exception_tasks1] PASSED [ 52%]
```

For the sake of making the test even simpler and consistent with other, I'd recommend one more thing. We can combine all `@pytest.mark.parametrize` together. It would look like: 

```suggestion
@pytest.mark.parametrize(
    "task, expectation", (
    (
        {
            "label": "task1",
            "dependencies": ["dependency"] * 2,
            "soft-dependencies": ["dependency"] * 1,
            "if-dependencies": ["dependency"] * 4,
        },
        does_not_raise(),
    ), (
    # ...
    ), (
        {
            "label": "task3",
            "dependencies": ["dependency"] * 98,
            "soft-dependencies": ["dependency"],
            "if-dependencies": ["dependency"],
        },
        pytest.raises(Exception),
    ),
))
```

This way, we don't have to use 2 `with` statements, a single will be able to deal with both cases (`does_not_raise()` and `pytest.raises(Exception)). Moreover, the expected result will live closer to the data under test which makes it easier to read byfuture readers. 

How does it sound to you? 🙂

Reply:
Yes! Sounds great, this would make the code shorter, cleaner and easier to read! Thank you for the suggestion! I will try to update it asap!

Reply:
I have made the necessary changes!